daemon/algod/api/server/v2/handlers.go

```diff
@@ -933,6 +939,218 @@ func (v2 *Handlers) TealDryrun(ctx echo.Context) error {
+		case catchup.ErrSyncModeNotEnabled:
+			return badRequest(ctx, err, errSyncModeNotEnabled, v2.Log)
+		case catchup.ErrSyncRoundInvalid:
+			return badRequest(ctx, err, errFailedSettingSyncRound, v2.Log)
+		default:
+			return internalError(ctx, err, errFailedSettingSyncRound, v2.Log)
+		}
+	}
+	return ctx.NoContent(http.StatusOK)
+}
+
+// GetSyncRound gets the sync round from the ledger.
+// (GET /v2/ledger/sync)
+func (v2 *Handlers) GetSyncRound(ctx echo.Context) error {
+	set, rnd, err := v2.Node.GetSyncRound()
+	if err != nil {
+		switch err {
+		case catchup.ErrSyncModeNotEnabled:
+			return badRequest(ctx, err, errSyncModeNotEnabled, v2.Log)
+		default:
+			return internalError(ctx, err, errFailedRetrievingSyncRound, v2.Log)
+		}
+	}
+	if !set {
+		return notFound(ctx, fmt.Errorf("sync round is not set"), errFailedRetrievingSyncRound, v2.Log)
+	}
+	return ctx.JSON(http.StatusOK, generated.GetSyncRoundResponse{Round: rnd})
+}
+
+// GetRoundDeltas returns the deltas for a given round.
+// This should be a combination of AccountDeltas, KVStore Deltas, etc.
+// (GET /v2/deltas/{round})
+func (v2 *Handlers) GetRoundDeltas(ctx echo.Context, round uint64) error {
+	ads, err := v2.Node.LedgerForAPI().GetAccountDeltasForRound(basics.Round(round))
+	if err != nil {
+		return internalError(ctx, err, errFailedRetrievingAccountDeltas, v2.Log)
+	}
+
+	var accts []generated.AccountBalanceRecord
+	var apps []generated.AppResourceRecord
+	var assets []generated.AssetResourceRecord
+
+	consensusParams, err := v2.Node.LedgerForAPI().ConsensusParams(basics.Round(round))
+	if err != nil {
+		return internalError(ctx, fmt.Errorf("unable to retrieve consensus params for round %d", round), errInternalFailure, v2.Log)
+	}
+	hdr, err := v2.Node.LedgerForAPI().BlockHdr(basics.Round(round))
+	if err != nil {
+		return internalError(ctx, fmt.Errorf("unable to retrieve block header for round %d", round), errInternalFailure, v2.Log)
+	}
+
+	for _, record := range ads.GetAllAccounts() {
+		var apiParticipation *generated.AccountParticipation
+		if record.VoteID != (crypto.OneTimeSignatureVerifier{}) {
+			apiParticipation = &generated.AccountParticipation{
+				VoteParticipationKey:      record.VoteID[:],
+				SelectionParticipationKey: record.SelectionID[:],
+				VoteFirstValid:            uint64(record.VoteFirstValid),
+				VoteLastValid:             uint64(record.VoteLastValid),
+				VoteKeyDilution:           uint64(record.VoteKeyDilution),
+			}
+			if !record.StateProofID.IsEmpty() {
+				tmp := record.StateProofID[:]
+				apiParticipation.StateProofKey = &tmp
+			}
+		}
+		var ot basics.OverflowTracker
+		pendingRewards := basics.PendingRewards(&ot, consensusParams, record.MicroAlgos, record.RewardsBase, hdr.RewardsLevel)
+
+		amountWithoutPendingRewards, overflowed := basics.OSubA(record.MicroAlgos, pendingRewards)
+		if overflowed {
+			return internalError(ctx, errors.New("overflow on pending reward calculation"), errInternalFailure, v2.Log)
+		}
```

Comment:
Where did this come from?

Reply:
https://github.com/algorand/go-algorand/blob/master/daemon/algod/api/server/v2/handlers.go#L444

The `generated.Account` type contains all of the information relevant to account deltas, so I didn't want to define a new type. The only thing it includes that looks unnecessary is the `AmountWithoutPendingRewards` field, but we're able to calculate it.